dhis-2/dhis-services/dhis-service-schema/src/main/java/org/hisp/dhis/schema/introspection/JacksonPropertyIntrospector.java

```diff
@@ -309,22 +311,12 @@ private static List<Property> collectProperties(Class<?> klass) {
       return Collections.emptyList();
     }
 
-    List<Field> fields =
-        ReflectionUtils.findFields(klass, f -> f.isAnnotationPresent(JsonProperty.class));
-
-    List<Method> methods =
-        ReflectionUtils.findMethods(
-            klass,
-            m ->
-                AnnotationUtils.findAnnotation(m, JsonProperty.class) != null
-                    && m.getParameterTypes().length == 0);
-
-    Multimap<String, Method> multimap = ReflectionUtils.getMethodsMultimap(klass);
+    List<Field> fields = findFields(klass, f -> f.isAnnotationPresent(JsonProperty.class));
 
     Map<String, Property> propertyMap = new HashMap<>();
 
     for (var field : fields) {
-      Property property = new Property(klass, null, null);
+      Property property = new Property(field.getType(), null, null);
```

Comment:
FYI: `klass` would be the type of the object having the `field` so it is the entirely wrong type for the property but it usually would be updated latter when a getter is found to the getter return type which is why it usually works unless you don't find the getter.